extensions/ticktick/src/utils/date.ts

```diff
@@ -34,15 +34,16 @@ export const formatToServerDate = (date: Date | moment.Moment | null | undefined
 };
 
 export const formatPrettyDateTime = (date: string, timezone: string, isFloating: boolean, isAllDay: boolean) => {
-  const datetimeInOriginalTimezone = moment.tz(date, timezone);
-
   if (!date) return "";
+  const dateInUTC = moment.utc(date);
+  if (!dateInUTC.isValid()) return "";
   else if (isAllDay) {
+    const datetimeInOriginalTimezone = moment.tz(dateInUTC, timezone);
     return datetimeInOriginalTimezone.format("MMM Do");
   } else if (isFloating) {
-    return moment(date).format("MMM Do, h:mm a");
+    return moment(dateInUTC).format("MMM Do, h:mm a");
```

Comment:
```typescript
if (isFloating) {
    const datetimeInOriginalTimezone = moment.tz(dateInUTC, timezone);
    return datetimeInOriginalTimezone.format("MMM Do, h:mm a");
}
```

Reply:
the floating time task displays the same date and time, so it should convert like an all-day task

Reply:
Ah thank you, updated now